exercises/luhn/test/Tests.hs

```diff
@@ -1,79 +1,73 @@
-        create 873956 `shouldBe` 8739567
-
-      it "create even larger valid number" $
-        create 837263756 `shouldBe` 8372637564
-
-    describe "track-specific tests" $ do
-
-      -- This track has some tests that were not included in the
-      -- reference test cases from `exercism/x-common/leap.json`.
-
-      it "checksum 1111" $
-        checksum 1111 `shouldBe` 6
-
-      it "checksum 8763" $
-        checksum 8763 `shouldBe` 0
-
-      it "checksum 8739567" $
-        checksum 8739567 `shouldBe` 0
-
-      it "checksum 2323200577663554" $
-        checksum 2323200577663554 `shouldBe` 0
-
-      it "isValid 1111" $
-        isValid 1111 `shouldBe` False
-
-      it "isValid 8763" $
-        isValid 8763 `shouldBe` True
-
-      it "isValid 2323200577663554" $
-        isValid 2323200577663554 `shouldBe` True
-
-      it "create 232320057766355" $
-        create 232320057766355 `shouldBe` 2323200577663554
+specs = describe "valid" $ for_ cases test
+  where
+    test Case{..} = it description $ isValid input `shouldBe` expected
+
+data Case = Case { description :: String
+                 , input       :: String
+                 , expected    :: Bool
+                 }
+
+cases :: [Case]
+cases = [ Case { description = "single digit strings can not be valid"
+               , input       = "1"
+               , expected    = False
+               }
+        , Case { description = "A single zero is invalid"
+               , input       = "0"
+               , expected    = False
+               }
+        , Case { description = "a simple valid SIN that remains valid if reversed"
+               , input       = "059"
+               , expected    = True
+               }
+        , Case { description = "a simple valid SIN that becomes invalid if reversed"
+               , input       = "59"
+               , expected    = True
+               }
+        , Case { description = "a valid Canadian SIN"
+               , input       = "055 444 285"
+               , expected    = True
+               }
+        , Case { description = "invalid Canadian SIN"
+               , input       = "055 444 286"
+               , expected    = False
+               }
+        , Case { description = "invalid credit card"
+               , input       = "8273 1232 7352 0569"
+               , expected    = False
+               }
+-- This track is not testing these cases, since we would rather focus on the algorithm,
+-- and because it seems strange to be unable to distinguish between well-formed invalid input and malformed input.
+--        , Case { description = "valid strings with a non-digit included become invalid"
+--               , input       = "055a 444 285"
+--               , expected    = False
+--               }
+--        , Case { description = "valid strings with punctuation included become invalid"
+--               , input       = "055-444-285"
+--               , expected    = False
+--               }
```

Comment:
Here it seems that you suppressed one test:

```json
    {
      "description": "valid strings with symbols included become invalid",
      "property": "valid",
      "input": "055£ 444$ 285",
      "expected": false
    },
```

Was it intentional?

I don't consider that a problem, because considering what was discussed in https://github.com/exercism/x-common/issues/428, it probably shouldn't exist anyway.